Dockerfile

```diff
@@ -5,88 +5,72 @@ ARG MUSL_TARGET="x86_64-linux-musl"
 # The crate features to build this with
 ARG FEATURES=""
 
-FROM --platform=$BUILDPLATFORM docker.io/alpine:latest as build
+FROM --platform=$BUILDPLATFORM rust:latest AS chef
 ARG RUST_TARGET
 ARG MUSL_TARGET
 ARG FEATURES
 
-RUN apk upgrade && \
-    apk add curl gcc musl-dev && \
-    curl -sSf https://sh.rustup.rs | sh -s -- --profile minimal --default-toolchain nightly --component rust-src -y
-
-RUN source $HOME/.cargo/env && \
-    mkdir -p /app/.cargo && \
-    if [ "$RUST_TARGET" != $(rustup target list --installed) ]; then \
-        rustup target add $RUST_TARGET && \
-        curl -L "https://musl.cc/$MUSL_TARGET-cross.tgz" -o /toolchain.tgz && \
-        tar xf toolchain.tgz && \
-        ln -s "/$MUSL_TARGET-cross/bin/$MUSL_TARGET-gcc" "/usr/bin/$MUSL_TARGET-gcc" && \
-        ln -s "/$MUSL_TARGET-cross/bin/$MUSL_TARGET-ld" "/usr/bin/$MUSL_TARGET-ld" && \
-        ln -s "/$MUSL_TARGET-cross/bin/$MUSL_TARGET-strip" "/usr/bin/actual-strip" && \
-        GCC_VERSION=$($MUSL_TARGET-gcc --version | grep gcc | awk '{print $3}') && \
-        echo -e "\
-[build]\n\
-rustflags = [\"-L\", \"native=/$MUSL_TARGET-cross/$MUSL_TARGET/lib\", \"-L\", \"native=/$MUSL_TARGET-cross/lib/gcc/$MUSL_TARGET/$GCC_VERSION/\", \"-l\", \"static=gcc\", \"-Z\", \"gcc-ld=lld\"]\n\
-[target.$RUST_TARGET]\n\
-linker = \"$MUSL_TARGET-gcc\"\n\
-[unstable]\n\
-build-std = [\"std\", \"panic_abort\"]\n\
-" > /app/.cargo/config; \
-    else \
-        echo "skipping toolchain as we are native" && \
-        echo -e "\
-[build]\n\
-rustflags = [\"-L\", \"native=/usr/lib\"]\n\
-[unstable]\n\
-build-std = [\"std\", \"panic_abort\"]\n\
-" > /app/.cargo/config && \
-        ln -s /usr/bin/strip /usr/bin/actual-strip; \
-    fi
+RUN <<EOT
+    set -ex
+    apt-get update
+    apt-get install --assume-yes musl-dev clang lld
+EOT
+
+RUN rustup target add $RUST_TARGET
+
+RUN cargo install cargo-chef --locked
+
+COPY <<EOF /app/.cargo/config
```

Comment:
the latest recommended way is to name the file `config.toml` indtead of just `config` but that might violate our msrv